
# Titanic regression

Identify which driver is the best predictor of survival: Ship class, Age, Sex,Fare paid?


In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
import sklearn

%matplotlib inline

In [3]:
#Import excel data set

location = "titanic3.xls"

dfTitanic = pd.read_excel(location)


dfTitanic.head()


pclass  survived                                             name     sex  \
0       1         1                    Allen, Miss. Elisabeth Walton  female   
1       1         1                   Allison, Master. Hudson Trevor    male   
2       1         0                     Allison, Miss. Helen Loraine  female   
3       1         0             Allison, Mr. Hudson Joshua Creighton    male   
4       1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

       age  sibsp  parch  ticket      fare    cabin embarked boat   body  \
0  29.0000      0      0   24160  211.3375       B5        S    2    NaN   
1   0.9167      1      2  113781  151.5500  C22 C26        S   11    NaN   
2   2.0000      1      2  113781  151.5500  C22 C26        S  NaN    NaN   
3  30.0000      1      2  113781  151.5500  C22 C26        S  NaN  135.0   
4  25.0000      1      2  113781  151.5500  C22 C26        S  NaN    NaN   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

# Data exploration

In [4]:
dfTitanic.columns

Index(['pclass', 'survived', 'name', 'sex', 'age', 'sibsp', 'parch', 'ticket',
       'fare', 'cabin', 'embarked', 'boat', 'body', 'home.dest'],
      dtype='object')

In [5]:
dfTitanic.dtypes

pclass         int64
survived       int64
name          object
sex           object
age          float64
sibsp          int64
parch          int64
ticket        object
fare         float64
cabin         object
embarked      object
boat          object
body         float64
home.dest     object
dtype: object

In [6]:
dfTitanic['sex'].value_counts()

male      843
female    466
Name: sex, dtype: int64

In [7]:
dfTitanic['ticket'].value_counts()

CA. 2343            11
1601                 8
CA 2144              8
3101295              7
347077               7
PC 17608             7
S.O.C. 14879         7
347082               7
347088               6
382652               6
113781               6
19950                6
4133                 5
113503               5
349909               5
PC 17757             5
W./C. 6608           5
220845               5
SC/Paris 2123        4
W./C. 6607           4
12749                4
113760               4
230136               4
C.A. 33112           4
16966                4
PC 17760             4
LINE                 4
PC 17483             4
17421                4
C.A. 34651           4
                    ..
113791               1
373450               1
A4. 54510            1
374910               1
315092               1
248659               1
368364               1
13568                1
2672                 1
349226               1
W./C. 14258          1
370372               1
2647       

In [8]:
dfTitanic['cabin'].value_counts()



C23 C25 C27        6
B57 B59 B63 B66    5
G6                 5
D                  4
B96 B98            4
F4                 4
F33                4
F2                 4
C78                4
C22 C26            4
A34                3
E34                3
B58 B60            3
C101               3
B51 B53 B55        3
E101               3
C32                2
D15                2
E46                2
C125               2
E44                2
B78                2
B35                2
D36                2
C80                2
C92                2
B20                2
E50                2
E25                2
D17                2
                  ..
F E46              1
C111               1
A32                1
C103               1
A29                1
F38                1
F E69              1
D47                1
B79                1
B38                1
B10                1
E68                1
C50                1
B86                1
F                  1
E77                1
C130         

In [9]:
dfTitanic['embarked'].value_counts()


S    914
C    270
Q    123
Name: embarked, dtype: int64

In [10]:
dfTitanic['boat'].value_counts()


13         39
C          38
15         37
14         33
4          31
10         29
5          27
3          26
9          25
11         25
7          23
8          23
16         23
6          20
D          20
12         19
2          12
A          11
B           9
1           5
5 7         2
C D         2
13 15       2
15 16       1
8 10        1
5 9         1
13 15 B     1
2           1
Name: boat, dtype: int64

In [11]:
dfTitanic['home.dest'].value_counts()


New York, NY                                          64
London                                                14
Montreal, PQ                                          10
Cornwall / Akron, OH                                   9
Paris, France                                          9
Wiltshire, England Niagara Falls, NY                   8
Philadelphia, PA                                       8
Winnipeg, MB                                           8
Brooklyn, NY                                           7
Belfast                                                7
Sweden Winnipeg, MN                                    7
Somerset / Bernardsville, NJ                           5
Ottawa, ON                                             5
Haverford, PA / Cooperstown, NY                        5
Bulgaria Chicago, IL                                   5
Sweden Worcester, MA                                   5
Rotherfield, Sussex, England Essex Co, MA              5
Bryn Mawr, PA                  

In [12]:
dfTitanic.isnull().sum()

pclass          0
survived        0
name            0
sex             0
age           263
sibsp           0
parch           0
ticket          0
fare            1
cabin        1014
embarked        2
boat          823
body         1188
home.dest     564
dtype: int64

In [14]:
dfTitanic.shape

(1309, 14)

In [15]:
dfTitanic['age'].value_counts()

24.0000    47
22.0000    43
21.0000    41
30.0000    40
18.0000    39
25.0000    34
28.0000    32
36.0000    31
27.0000    30
29.0000    30
26.0000    30
19.0000    29
23.0000    26
32.0000    24
20.0000    23
35.0000    23
31.0000    23
45.0000    21
33.0000    21
17.0000    20
39.0000    20
16.0000    19
40.0000    18
42.0000    18
34.0000    16
50.0000    15
47.0000    14
48.0000    14
38.0000    14
2.0000     12
           ..
28.5000     3
18.5000     3
12.0000     3
71.0000     2
45.5000     2
36.5000     2
34.5000     2
14.5000     2
0.9167      2
30.5000     2
70.0000     2
55.5000     1
20.5000     1
0.4167      1
60.5000     1
23.5000     1
22.5000     1
38.5000     1
66.0000     1
24.5000     1
74.0000     1
0.1667      1
0.3333      1
80.0000     1
11.5000     1
26.5000     1
67.0000     1
70.5000     1
76.0000     1
0.6667      1
Name: age, Length: 98, dtype: int64

In [16]:
dfTitanic['body'].value_counts()

259.0    1
305.0    1
51.0     1
68.0     1
47.0     1
98.0     1
188.0    1
120.0    1
156.0    1
37.0     1
58.0     1
70.0     1
196.0    1
61.0     1
53.0     1
4.0      1
7.0      1
32.0     1
176.0    1
50.0     1
1.0      1
14.0     1
312.0    1
328.0    1
304.0    1
297.0    1
283.0    1
285.0    1
327.0    1
142.0    1
        ..
101.0    1
121.0    1
292.0    1
299.0    1
286.0    1
322.0    1
294.0    1
209.0    1
187.0    1
153.0    1
255.0    1
173.0    1
275.0    1
287.0    1
261.0    1
258.0    1
201.0    1
230.0    1
271.0    1
234.0    1
245.0    1
166.0    1
109.0    1
149.0    1
174.0    1
97.0     1
197.0    1
130.0    1
75.0     1
135.0    1
Name: body, Length: 121, dtype: int64

In [17]:
dfTitanic.max()

pclass                                3
survived                              1
name        van Melkebeke, Mr. Philemon
sex                                male
age                                  80
sibsp                                 8
parch                                 9
fare                            512.329
body                                328
dtype: object

In [18]:
dfTitanic.min()

pclass                        1
survived                      0
name        Abbing, Mr. Anthony
sex                      female
age                      0.1667
sibsp                         0
parch                         0
fare                          0
body                          1
dtype: object

In [19]:
dfTitanic.apply(pd.Series.value_counts)

C:\Users\EDM115\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\indexes\api.py:107: RuntimeWarning: '<' not supported between instances of 'int' and 'str', sort order is undefined for incomparable objects
  result = result.union(other)
C:\Users\EDM115\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\frame.py:7397: RuntimeWarning: '<' not supported between instances of 'int' and 'str', sort order is undefined for incomparable objects
  index = _union_indexes(indexes)
C:\Users\EDM115\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\indexes\api.py:107: RuntimeWarning: '<' not supported between instances of 'str' and 'int', sort order is undefined for incomparable objects
  result = result.union(other)


pclass  survived  name  \
0                                                      NaN     809.0   NaN   
1                                                    323.0     500.0   NaN   
2                                                    277.0       NaN   NaN   
3                                                    709.0       NaN   NaN   
Connolly, Miss. Kate                                   NaN       NaN   2.0   
Kelly, Mr. James                                       NaN       NaN   2.0   
Denbury, Mr. Herbert                                   NaN       NaN   1.0   
Palsson, Master. Gosta Leonard                         NaN       NaN   1.0   
Woolner, Mr. Hugh                                      NaN       NaN   1.0   
Gilinski, Mr. Eliezer                                  NaN       NaN   1.0   
Lennon, Mr. Denis                                      NaN       NaN   1.0   
Yousseff, Mr. Gerious                                  NaN       NaN   1.0   
Dorking, Mr. Edward Arthur                             NaN       NaN   1.0   
Caram, Mr. Joseph                                      NaN       NaN   1.0   
Schabert, Mrs. Paul (Emma Mock)                        NaN       NaN   1.0   
Mudd, Mr. Thomas Charles                               NaN       NaN   1.0   
Elsbury, Mr. William James                             NaN       NaN   1.0   
Astor, Mrs. John Jacob (Madeleine Talmadge Force)      NaN       NaN   1.0   
Dimic, Mr. Jovan                                       NaN       NaN   1.0   
Borebank, Mr. John James                               NaN       NaN   1.0   
Lindstrom, Mrs. Carl Johan (Sigrid Posse)              NaN       NaN   1.0   
Frost, Mr. Anthony Wood "Archie"                       NaN       NaN   1.0   
Ivanoff, Mr. Kanio                                     NaN       NaN   1.0   
Peacock, Mrs. Benjamin (Edith Nile)                    NaN       NaN   1.0   
Boulos, Mrs. Joseph (Sultana)                          NaN       NaN   1.0   
Baclini, Miss. Helene Barbara                          NaN       NaN   1.0   
Williams, Mr. Leslie                                   NaN       NaN   1.0   
Lines, Mrs. Ernest H (Elizabeth Lindsey James)         NaN       NaN   1.0   
Andersson, Mr. August Edvard ("Wennerstrom")           NaN       NaN   1.0   
Cann, Mr. Ernest Charles                               NaN       NaN   1.0   
...                                                    ...       ...   ...   
Dorchester, MA                                         NaN       NaN   NaN   
Medeltorp, Sweden Chicago, IL                          NaN       NaN   NaN   
Provo, UT                                              NaN       NaN   NaN   
Cologne, Germany                                       NaN       NaN   NaN   
Kilmacowen, Co Sligo, Ireland New York, NY             NaN       NaN   NaN   
Rotterdam, Netherlands                                 NaN       NaN   NaN   
Kontiolahti, Finland / Detroit, MI                     NaN       NaN   NaN   
Wimbledon Park, London / Hayling Island, Hants         NaN       NaN   NaN   
Paris, France New York, NY                             NaN       NaN   NaN   
Bath, England / Massachusetts                          NaN       NaN   NaN   
Tokyo, Japan                                           NaN       NaN   NaN   
Albany, NY                                             NaN       NaN   NaN   
Bryn Mawr, PA, USA                                     NaN       NaN   NaN   
Clevedon, England                                      NaN       NaN   NaN   
Deer Lodge, MT                                         NaN       NaN   NaN   
Bournemouth, England                                   NaN       NaN   NaN   
Plymouth, Dorset / Houghton, MI                        NaN       NaN   NaN   
Bayside, Queens, NY                                    NaN       NaN   NaN   
England New York, NY                                   NaN       NaN   NaN   
Norway Los Angeles, CA                                 NaN       NaN   NaN

In [20]:
dfTitanic['survived'].value_counts()

0    809
1    500
Name: survived, dtype: int64

In [21]:
print(dfTitanic.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 14 columns):
pclass       1309 non-null int64
survived     1309 non-null int64
name         1309 non-null object
sex          1309 non-null object
age          1046 non-null float64
sibsp        1309 non-null int64
parch        1309 non-null int64
ticket       1309 non-null object
fare         1308 non-null float64
cabin        295 non-null object
embarked     1307 non-null object
boat         486 non-null object
body         121 non-null float64
home.dest    745 non-null object
dtypes: float64(3), int64(4), object(7)
memory usage: 143.2+ KB
None


In [36]:
dfTitanic.rename(columns={"home.dest":"home"}, inplace= True)
dfTitanic.columns

Index(['pclass', 'survived', 'name', 'sex', 'age', 'sibsp', 'parch', 'ticket',
       'fare', 'cabin', 'embarked', 'boat', 'body', 'home'],
      dtype='object')

In [40]:
dfTitanic.home.unique()

array(['St Louis, MO', 'Montreal, PQ / Chesterville, ON', 'New York, NY',
       'Hudson, NY', 'Belfast, NI', 'Bayside, Queens, NY',
       'Montevideo, Uruguay', 'Paris, France', nan, 'Hessle, Yorks',
       'Montreal, PQ', 'Winnipeg, MN', 'San Francisco, CA',
       'Dowagiac, MI', 'Stockholm, Sweden / Washington, DC',
       'Trenton, NJ', 'Glen Ridge, NJ', 'Youngstown, OH',
       'Birkdale, England Cleveland, Ohio', 'London / Winnipeg, MB',
       'Cooperstown, NY', 'St Leonards-on-Sea, England Ohio',
       'Los Angeles, CA', 'Pomeroy, WA', 'Omaha, NE', 'Philadelphia, PA',
       'Denver, CO', 'Belmont, MA', 'Washington, DC',
       'Austria-Hungary / Germantown, Philadelphia, PA',
       'Germantown, Philadelphia, PA', 'Bryn Mawr, PA',
       'Ascot, Berkshire / Rochester, NY', 'Little Onn Hall, Staffs',
       'Amenia, ND', 'New York, NY / Ithaca, NY', 'London, England',
       'Liverpool, England / Belfast', 'Stoughton, MA', 'Victoria, BC',
       'Lakewood, NJ', 'Roachdale, I

In [41]:
dfTitanic.body.unique()

array([ nan, 135.,  22., 124., 148., 208., 172., 269.,  62., 133., 275.,
       147., 110., 307.,  38.,  80.,  45., 258., 126., 292., 175., 249.,
       230., 122., 263., 234., 189., 166., 207., 232.,  16., 109.,  96.,
        46., 245., 169., 174.,  97.,  18., 130.,  17., 295., 286., 236.,
       322., 297., 155., 305.,  19.,  75.,  35., 256., 149., 283., 165.,
       108., 121.,  52., 209., 271.,  43.,  15., 101., 287.,  81., 294.,
       293., 190.,  72., 103.,  79., 259., 260., 142., 299., 171.,   9.,
       197.,  51., 187.,  68.,  47.,  98., 188.,  69., 306., 120., 143.,
       156., 285.,  37.,  58.,  70., 196., 153.,  61.,  53., 201., 309.,
       181., 173.,  89.,   4., 206., 327., 119.,   7.,  32.,  67., 284.,
       261., 176.,  50.,   1., 255., 298., 314.,  14., 131., 312., 328.,
       304.])

In [42]:
dfTitanic.boat.unique()

array([2, '11', nan, '3', '10', 'D', '4', '9', '6', 'B', '8', 'A', '5',
       '7', 'C', '14', '2', '5 9', '13', '1', '15', '5 7', '8 10', '12',
       '16', '13 15 B', 'C D', '15 16', '13 15'], dtype=object)

In [43]:
dfTitanic.embarked.unique()

array(['S', 'C', nan, 'Q'], dtype=object)

In [44]:
dfTitanic.cabin.unique()

array(['B5', 'C22 C26', 'E12', 'D7', 'A36', 'C101', nan, 'C62 C64', 'B35',
       'A23', 'B58 B60', 'D15', 'C6', 'D35', 'C148', 'C97', 'B49', 'C99',
       'C52', 'T', 'A31', 'C7', 'C103', 'D22', 'E33', 'A21', 'B10', 'B4',
       'E40', 'B38', 'E24', 'B51 B53 B55', 'B96 B98', 'C46', 'E31', 'E8',
       'B61', 'B77', 'A9', 'C89', 'A14', 'E58', 'E49', 'E52', 'E45',
       'B22', 'B26', 'C85', 'E17', 'B71', 'B20', 'A34', 'C86', 'A16',
       'A20', 'A18', 'C54', 'C45', 'D20', 'A29', 'C95', 'E25', 'C111',
       'C23 C25 C27', 'E36', 'D34', 'D40', 'B39', 'B41', 'B102', 'C123',
       'E63', 'C130', 'B86', 'C92', 'A5', 'C51', 'B42', 'C91', 'C125',
       'D10 D12', 'B82 B84', 'E50', 'D33', 'C83', 'B94', 'D49', 'D45',
       'B69', 'B11', 'E46', 'C39', 'B18', 'D11', 'C93', 'B28', 'C49',
       'B52 B54 B56', 'E60', 'C132', 'B37', 'D21', 'D19', 'C124', 'D17',
       'B101', 'D28', 'D6', 'D9', 'B80', 'C106', 'B79', 'C47', 'D30',
       'C90', 'E38', 'C78', 'C30', 'C118', 'D36', 'D48', 'D47', '

In [45]:
dfTitanic.fare.unique()

array([211.3375, 151.55  ,  26.55  ,  77.9583,   0.    ,  51.4792,
        49.5042, 227.525 ,  69.3   ,  78.85  ,  30.    ,  25.925 ,
       247.5208,  76.2917,  75.2417,  52.5542, 221.7792,  26.    ,
        91.0792, 135.6333,  35.5   ,  31.    , 164.8667, 262.375 ,
        55.    ,  30.5   ,  50.4958,  39.6   ,  27.7208, 134.5   ,
        26.2875,  27.4458, 512.3292,   5.    ,  47.1   , 120.    ,
        61.175 ,  53.1   ,  86.5   ,  29.7   , 136.7792,  52.    ,
        25.5875,  83.1583,  25.7   ,  71.    ,  71.2833,  57.    ,
        81.8583, 106.425 ,  56.9292,  78.2667,  31.6792,  31.6833,
       110.8833,  26.3875,  27.75  , 263.    , 133.65  ,  49.5   ,
        79.2   ,  38.5   , 211.5   ,  59.4   ,  89.1042,  34.6542,
        28.5   , 153.4625,  63.3583,  55.4417,  76.7292,  42.4   ,
        83.475 ,  93.5   ,  42.5   ,  51.8625,  50.    ,  57.9792,
        90.    ,  30.6958,  80.    ,  28.7125,  25.9292,  39.4   ,
        45.5   , 146.5208,  82.1708,  57.75  , 113.275 ,  26.2

In [46]:
dfTitanic.ticket.unique()

array([24160, 113781, 19952, 13502, 112050, 11769, 'PC 17609', 'PC 17757',
       'PC 17477', 19877, 27042, 'PC 17318', 'PC 17558', 11813, 13050,
       11751, 111369, 'PC 17483', 13905, 11967, 'PC 17760', 110564,
       113784, 112277, 36928, 113783, 110489, 'PC 17608', 113505, 111427,
       113054, 'PC 17591', 112379, 'PC 17610', 16966, 113050, 113798,
       'PC 17476', 'PC 17606', 'PC 17755', 695, 113059, 113760, 19924,
       17770, 'W.E.P. 5734', 113806, 110152, 'PC 17594', 112051, 13508,
       110465, 5727, 'PC 17756', 11770, 113791, 'WE/P 5735', 112901,
       'PC 17599', 113055, 113804, 'F.C. 12750', 17474, 33638, 'PC 17761',
       11755, 'PC 17485', 'PC 17580', 11767, 36947, 'PC 17531',
       'PC 17598', '17421', 'PC 17474', '113051', '19950', '113778',
       'PC 17611', '17765', '13568', '13567', '112058', '113803',
       '111320', '113503', '112378', 'PC 17593', '17453', 'PC 17754',
       '113780', '112053', 'PC 17582', 'PC 17759', '11765', 'PC 17572',
       '113796

In [47]:
dfTitanic.parch.unique()

array([0, 2, 1, 4, 3, 5, 6, 9], dtype=int64)

In [48]:
dfTitanic.sibsp.unique()

array([0, 1, 2, 3, 4, 5, 8], dtype=int64)

In [49]:
dfTitanic.age.unique()

array([29.    ,  0.9167,  2.    , 30.    , 25.    , 48.    , 63.    ,
       39.    , 53.    , 71.    , 47.    , 18.    , 24.    , 26.    ,
       80.    ,     nan, 50.    , 32.    , 36.    , 37.    , 42.    ,
       19.    , 35.    , 28.    , 45.    , 40.    , 58.    , 22.    ,
       41.    , 44.    , 59.    , 60.    , 33.    , 17.    , 11.    ,
       14.    , 49.    , 76.    , 46.    , 27.    , 64.    , 55.    ,
       70.    , 38.    , 51.    , 31.    ,  4.    , 54.    , 23.    ,
       43.    , 52.    , 16.    , 32.5   , 21.    , 15.    , 65.    ,
       28.5   , 45.5   , 56.    , 13.    , 61.    , 34.    ,  6.    ,
       57.    , 62.    , 67.    ,  1.    , 12.    , 20.    ,  0.8333,
        8.    ,  0.6667,  7.    ,  3.    , 36.5   , 18.5   ,  5.    ,
       66.    ,  9.    ,  0.75  , 70.5   , 22.5   ,  0.3333,  0.1667,
       40.5   , 10.    , 23.5   , 34.5   , 20.5   , 30.5   , 55.5   ,
       38.5   , 14.5   , 24.5   , 60.5   , 74.    ,  0.4167, 11.5   ,
       26.5   ])

In [50]:
dfTitanic.sex.unique()

array(['female', 'male'], dtype=object)

In [51]:
dfTitanic.survived.unique()

array([1, 0], dtype=int64)

In [52]:
dfTitanic.pclass.unique()

array([1, 2, 3], dtype=int64)

# Droping fields

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 14 columns):
pclass       1309 non-null int64->Include
survived     1309 non-null int64->Include
name         1309 non-null object->Drop
sex          1309 non-null object->Convert Include
age          1046 non-null float64->Include
sibsp        1309 non-null int64->Include
parch        1309 non-null int64->Include
ticket       1309 non-null object->Drop
fare         1308 non-null float64->Include
cabin        295 non-null object->Drop
embarked     1307 non-null object-> Convert Include
boat         486 non-null object-> Convert Include Drop
body         121 non-null float64->Include
home         745 non-null object->Drop
dtypes: float64(3), int64(4), object(7)
memory usage: 143.2+ KB
None



In [54]:
dfTitanic1= dfTitanic.drop(['name','ticket','cabin','boat','home'],axis=1)
dfTitanic1.columns



Index(['pclass', 'survived', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked', 'body'],
      dtype='object')

# Converting obj -> int data type/ Clean nan
1.Identify list of unique values
2.Create a dictionary
3. Replace obj data type values with int data type values using dic
4. A function could also be used to replace values

In [55]:
#sex Creade dic array(['female', 'male'])

sexNum={'female':1, 'male':0}

In [56]:
#sex Replace values using dic

dfTitanic1= dfTitanic1.replace({'sex':sexNum})


In [58]:
#embarked array(['S', 'C', nan, 'Q'])

embarkedNum={'S':1, 'C':2, 'Q':3}


In [59]:
#embarked replace values using dic

dfTitanic1 = dfTitanic1.replace({'embarked':embarkedNum})


In [60]:
#Status of nan values

dfTitanic1.isnull().sum()

pclass         0
survived       0
sex            0
age          263
sibsp          0
parch          0
fare           1
embarked       2
body        1188
dtype: int64

In [61]:
#Clean data
dfTitanic2= dfTitanic1.drop(['body'],axis=1)
#dfTitanic2.columns


In [65]:
dfTitanic3= dfTitanic2.dropna()

In [66]:
dfTitanic3.columns

Index(['pclass', 'survived', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked'],
      dtype='object')

In [67]:
dfTitanic3.isnull().sum()

pclass      0
survived    0
sex         0
age         0
sibsp       0
parch       0
fare        0
embarked    0
dtype: int64

In [68]:
#Drop dependable variable; 
from sklearn.linear_model import LinearRegression
X = dfTitanic3.drop(['survived'], axis = 1)

#assign linear regression function to a variable
lm = LinearRegression()
lm




LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [69]:
lm.fit(X, dfTitanic3.survived)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [70]:
print('Estimated intercept coefficient:', lm.intercept_)

Estimated intercept coefficient: 0.7783708515094567


In [71]:
print('Number of coefficients:', len(lm.coef_))

Number of coefficients: 7


In [72]:
pd.DataFrame(list(zip(X.columns, lm.coef_)), columns = ['features', 'estimatedCoefficients'])

features  estimatedCoefficients
0    pclass              -0.175679
1       sex               0.488680
2       age              -0.005980
3     sibsp              -0.051667
4     parch               0.010935
5      fare               0.000221
6  embarked               0.021511